include/crab/domains/inter_abstract_operations_callsite_info.hpp

```diff
@@ -17,18 +17,30 @@ template <class Variable> class callsite_info {
   const std::vector<Variable> &m_caller_out_params;
   const std::vector<Variable> &m_callee_in_params;
   const std::vector<Variable> &m_callee_out_params;
+  const std::vector<std::vector<Variable>> &m_caller_in_groups;
```

Comment:
I wonder if there is something better than a vector of vectors. 
I know that you want to have explicitly each equivalence class.
But it doesn't look to me very elegant but maybe that's just personal taste. 
An alternative would be just to keep a map from Variable to Variable (similar to what an union-find would keep internally). Eg.., `{{x,y}{p,q}}` can be represented with the map `{x -> x, y -> x, p -> p, q -> p}`. This map is good to answer questions about whether two variables are in the same equivalence class but you would need to create the vector of vectors probably anyway.

Reply:
I agree the map could help us identify whether two variables belong to the same equivalence class. However, I don't need this map for the `extend` and `restrict` methods since an object copy is performed by copying subdomains and renaming them through a vector of variables. Using a vector of vectors is the simplest way to represent equivalence classes. I prefer maintaining a vector of vectors at this point.